src/main/scala/leon/comparison/ComparatorExprList.scala

```diff
@@ -0,0 +1,108 @@
+package leon.comparison
+
+import leon.comparison.Utils._
+import leon.purescala.Expressions.{CaseClassPattern, _}
+
+/**
+  * Created by joachimmuth on 25.04.16.
+  *
+  * This way of basic comparison flat both functional trees into lists and compare them in every possible combination.
+  *
+  * The easy-to-understand way of working provides a point of comparison for further advanced method.
+  *
+  * For clarity, we always pass "corpus function" first in argument
+  */
+object ComparatorExprList extends Comparator {
+  val name = "ExprList"
+
+  def compare(expr_corpus: Expr, expr: Expr)= {
+    val exprsA = collectExpr(expr_corpus)
+    val exprsB = collectExpr(expr)
+
+    val numberOfSimilarExpr: Int = pairsOfSimilarExp(exprsA, exprsB)
+
+    val score = Utils.matchScore(numberOfSimilarExpr, exprsA.size, exprsB.size)
+
+    if (score > 0.0 && ComparisonPhase.debug){
+      println("---------------------")
+      println("COMPARATOR " + name)
+      println("Expressions: ", expr_corpus, expr)
+      println("similar expr: ", numberOfSimilarExpr)
+      println("---------------------")
+    }
+
+    (score, "")
+  }
+
+
+  def pairsOfSimilarExp(exprs_corpus: List[Expr], exprs: List[Expr]): Int = {
+    def helper(exprs_corpus: List[Expr], exprs: List[Expr], acc: Int): Int = exprs match {
```

Comment:
what I wanted was in fact List.intersect, because I wanted that:
List(1, 1) and List(1) give me 1
List(1,1) and List (1, 1) give me 2